src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -161,10 +167,11 @@ public ParseStatus parseJavadocAsDetailNode(DetailAST javadocCommentAst) {
      * Parses block comment content as javadoc comment.
      * @param blockComment
      *        block comment content.
+     * @param result Reference to object returned by {@link #parseJavadocAsDetailNode(DetailAST)}
      * @return parse tree
      * @noinspection deprecation
      */
-    private ParseTree parseJavadocAsParseTree(String blockComment) {
+    private ParseTree parseJavadocAsParseTree(String blockComment, ParseStatus result) {
```

Comment:
1. Maybe.

2. I guess, It will not be thread-safe as long we use one instance of `JavadocDetailNodeParser` in `AbstractJavadocCheck`.
Strange that `JavadocDetailNodeParser` has other class members: `blockCommentLineNumber` and errorListener. And even stranger why do we use only one instance of `JavadocDetailNodeParser` in `AbstractJavadocCheck`.

@romani, please share your thoughts